cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -436,30 +428,47 @@ func (o *ScaleUpOrchestrator) ComputeExpansionOption(
 	nodeGroup cloudprovider.NodeGroup,
 	nodeInfo *schedulerframework.NodeInfo,
 	upcomingNodes []*schedulerframework.NodeInfo,
-) (expander.Option, error) {
+) expander.Option {
 	option := expander.Option{
 		NodeGroup: nodeGroup,
 		Pods:      make([]*apiv1.Pod, 0),
 	}
 
+	option.Pods = o.SchedulablePods(podEquivalenceGroups, nodeGroup, nodeInfo)
+	if len(option.Pods) > 0 {
+		estimator := o.autoscalingContext.EstimatorBuilder(o.autoscalingContext.PredicateChecker, o.autoscalingContext.ClusterSnapshot)
+		option.NodeCount, option.Pods = estimator.Estimate(option.Pods, nodeInfo, option.NodeGroup)
+	}
+
+	return option
+}
+
+// SchedulablePods returns a list of pods that could be scheduled
+// in a given node group after a scale up.
+func (o *ScaleUpOrchestrator) SchedulablePods(
+	podEquivalenceGroups []*equivalence.PodGroup,
+	nodeGroup cloudprovider.NodeGroup,
+	nodeInfo *schedulerframework.NodeInfo,
+) []*apiv1.Pod {
 	o.autoscalingContext.ClusterSnapshot.Fork()
+	defer o.autoscalingContext.ClusterSnapshot.Revert()
 
 	// Add test node to snapshot.
-	var pods []*apiv1.Pod
+	var allPods []*apiv1.Pod
 	for _, podInfo := range nodeInfo.Pods {
-		pods = append(pods, podInfo.Pod)
+		allPods = append(allPods, podInfo.Pod)
 	}
-	if err := o.autoscalingContext.ClusterSnapshot.AddNodeWithPods(nodeInfo.Node(), pods); err != nil {
+	if err := o.autoscalingContext.ClusterSnapshot.AddNodeWithPods(nodeInfo.Node(), allPods); err != nil {
 		klog.Errorf("Error while adding test Node: %v", err)
-		o.autoscalingContext.ClusterSnapshot.Revert()
-		return expander.Option{}, nil
+		return []*apiv1.Pod{}
 	}
 
+	schedulablePods := make([]*apiv1.Pod, 0)
```

Comment:
micro nit: Why not just `var schedulablePods []*apiv1.Pod`?

Reply:
This code is just moved to a dedicated function (so I do not know)
Either way I applied the suggestion :)